### ISSUE TITLE
400 bad request on get

### ISSUE THREAD

**Issue by Chaz6** [ducaale/ht#20]
`curl` gives the expected response, whilst `ht` does not.

```
$ curl -i http://chaz6.com
HTTP/1.1 301 Moved Permanently
Server: nginx
Date: Sat, 06 Feb 2021 12:42:53 GMT
Content-Type: text/html
Content-Length: 162
Connection: keep-alive
Location: https://chaz6.com/

<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>

$ ht get http://chaz6.com
HTTP/2.0 400 Bad Request
content-length: 150
content-type: text/html
date: Sat, 06 Feb 2021 12:41:40 GMT
server: nginx

<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

**Comment by Chaz6** [comment]
Some additional output:-

```
$ curl -i --http2 https://chaz6.com

HTTP/2 200
server: nginx
date: Sat, 06 Feb 2021 12:49:38 GMT
content-type: text/html
content-length: 4271
last-modified: Tue, 18 Feb 2020 21:09:10 GMT
etag: "5e4c5276-10af"
[..]

$ ht get https://chaz6.com
HTTP/2.0 400 Bad Request
content-length: 150
content-type: text/html
date: Sat, 06 Feb 2021 12:50:33 GMT
server: nginx

<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

**Comment by Till--H** [comment]
The difference from the first comment comes from a difference in the handling of redirects and should be fixed by https://github.com/ducaale/ht/pull/19; the second comment has likely the same root cause as https://github.com/ducaale/ht/issues/13 (i.e., handling of HTTP2)

**Comment by ducaale** [comment]
This is now fixed as part of the ht v0.5.0 release. Feel free to reopen the issue if the bug still persists.
